swap_meet/vendor.py

```diff
@@ -1,2 +1,92 @@
+from .item import Item
+from .decor import Decor
+from .electronics import Electronics
+from .clothing import Clothing
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item):
+       self.inventory.append(item)
+       return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else:
+           self.inventory.remove(item)
+           return item
+        
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+            if isinstance(id, Item):
+                raise ValueError("id should be an int!")
+
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        if their_item not in self.inventory and my_item not in other_vendor.inventory:
+            my_index, their_index = self.inventory.index(my_item), other_vendor.inventory.index(their_item)
+            self.inventory[my_index], other_vendor.inventory[their_index] = their_item, my_item
+            return True
+        return False
+        
+        
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        self.inventory[0], other_vendor.inventory[0] = (other_vendor.inventory[0],                                                     self.inventory[0])
+        return True
+        
+    def get_by_category(self, category):
+        matching_items = []
+        matching_items = [item for item in self.inventory if item.get_category() == category]
+        if matching_items:
+            return matching_items
+        else:
+            return []
+    
+    def get_best_by_category(self, category):
+        items = self.get_by_category(category)
+        if not items:
+            return None
+        return max(items, key=lambda item: item.condition)
+    
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        my_best_item = self.get_best_by_category(their_priority)
+        their_best_item = other_vendor.get_best_by_category(my_priority)
+        if my_best_item and their_best_item:
```

Comment:
Could we change this into a guard statement to let us unindent the intended flow of the function?